html/exercise-15/5e.js

```diff
@@ -0,0 +1,137 @@
+const cards = document.querySelectorAll('.card');
+const sidePanel = document.querySelector('.side-panel');
+const infoIframe = document.querySelector('#info-iframe');
+const moreButtons = document.querySelectorAll('.more-button');
+
+// moreButtons.forEach(button => {
+//   button.addEventListener('click', (event) => {
+//     const movieIndex = event.target.getAttribute('data-movie-index');
+//     const movieInfo = getMovieInfo(movieIndex); // Replace with your data retrieval logic
+//     updateIframe(movieInfo);
+//     sidePanel.style.right = '0';
+//   });
+// });
+
+// function updateIframe(info) {
+//   const iframeContent = `
+//     <h2>${info.title}</h2>
+//     <p>Release Date: ${info.releaseDate}</p>
+//     <p>Plot: ${info.plot}</p>
+//     <p>Ratings: ${info.ratings.join(', ')}</p>
+//     <p>Languages: ${info.languages.join(', ')}</p>
+//   `;
+//   infoIframe.contentDocument.body.innerHTML = iframeContent;
+// }
+
+// function getMovieInfo(index) {
+//   // Replace this with your data retrieval logic
+//   return {
+//     title: `Movie Title ${index}`,
+//     releaseDate: `2023-08-2${index}`,
+//     plot: `Plot for Movie ${index}`,
+//     ratings: ['Rating 1', 'Rating 2'],
+//     languages: ['Language 1', 'Language 2'],
+//   };
+// }
+
+function show(data) {
+  var url="https://chat.openai.com/";
+  var moviesContainer = document.getElementById("movies-container");
+
+  var moviesHTML = "";
+
+  var movies = data.Search;
+  for (var i = 0; i < movies.length; i++) {
+      console.log(movies[i].Title);
+
+      var movie = movies[i];
+      
+      moviesHTML += `
+          <div class="movie">
+          <img src="${movie.Poster}" alt="${movie.Title} Poster">
+              <h4 class="title">${movie.Title}</h4>
+              <p class="year">Year: ${movie.Year}</p>
+              <p class="type">Type: ${movie.Type}</p>
+              <button class"view-more" onclick="viewMore('${movie.Title}')">View more</button>
+          </div>
+      `;
+  }
+
+
+  moviesContainer.innerHTML = moviesHTML;
+
+}
+
+function viewMore(title) {
+  var url = "https://www.omdbapi.com/?apikey=6b40f680&t=" + encodeURIComponent(title);
```

Comment:
Ideally, the api key should be stored away in an env file, since it's used in multiple places consider creating a new variable.